src/general/bprime.cpp

```diff
@@ -54,38 +56,231 @@ using namespace Mutation::Thermodynamics;
+            dx = atof(tokens[1].c_str());
+            break;
+        default:
+            return false;
+    }
+
+    if (dx == 0.0) {
+        x2 = x1;
+        dx = 1.0;
+    }
+
+    return true;
+}
+
+
+bool parseRange(const std::string& range, double& x1)
+{
+    std::vector<std::string> tokens;
+    String::tokenize(range, tokens, ":");
+
+    if (!String::isNumeric(tokens))
+        return false;
+
+    x1 = atof(tokens[0].c_str());
+
+
+    return true;
+}
+
+// Parse the command line options to determine what the user wants to do
+Options parseOptions(int argc, char** argv)
+{
+    Options opts;
+
+    // Print the help message and exit if desired
+    if(argc<2)
+        printHelpMessage(argv[0]);
+
+    if (optionExists(argc, argv, "-h") || optionExists(argc, argv, "--help"))
+        printHelpMessage(argv[0]);
+
+    // Get the temperature range
+    if (optionExists(argc, argv, "-T")) {
+        if (!parseRange(
+                getOption(argc, argv, "-T"), opts.T1, opts.T2, opts.dT)) {
+            cout << "Bad format for temperature range!" << endl;
+            printHelpMessage(argv[0]);
+        }
+    } else {
+        opts.T1 = 300.0;
+        opts.T2 = 15000.0;
+        opts.dT = 100.0;
+    }
+
+    // Get the pressure range
+    if (optionExists(argc, argv, "-P")) {
+        if (!parseRange(
+                getOption(argc, argv, "-P"), opts.P1)) {
+            cout << "Bad format for pressure !" << endl;
+            printHelpMessage(argv[0]);
+        }
+    } else {
+        opts.P1 = ONEATM;
+    }
+
+    // Get the pressure range
+    if (optionExists(argc, argv, "-b")) {
+        if (!parseRange(
+                getOption(argc, argv, "-b"), opts.Bg)) {
+            cout << "Bad format for pressure !" << endl;
+            printHelpMessage(argv[0]);
+        }
+    } else {
+        opts.Bg = 0;
+    }
+
+    if (optionExists(argc, argv, "-m")) {
+        opts.mixture = getOption(argc, argv, "-m");
+    } else {
+        printHelpMessage(argv[0]);
+    }
+
+
+    if (optionExists(argc, argv, "-bl")) {
+        opts.boundary_layer_comp = getOption(argc, argv, "-bl");
+    } else {
+        printHelpMessage(argv[0]);
+    }
+
+    if (optionExists(argc, argv, "-py")) {
+        opts.pyrolysis_composition = getOption(argc, argv, "-py");
+        opts.pyrolysis_exist= true;
+    }
+
+    return opts;
+}
+
 
 int main(int argc, char* argv[])
 {
 #ifdef _GNU_SOURCE
     feenableexcept(FE_INVALID | FE_DIVBYZERO | FE_OVERFLOW);
 #endif
 
-    if (argc != 9) {
-        cout << "Usage: bprime T1 T2 dT P bg mixture BL_comp Pyro_comp" << endl;
-        exit(1);
-    }
 
-    Mixture mix(argv[6]);
+    Options opts = parseOptions(argc, argv);
+
+    Mixture mix(opts.mixture);
     
     const int ne = mix.nElements();
     const int ns = mix.nSpecies();
     
-    double* p_Yke = new double [ne];
-    double* p_Ykg = new double [ne];
-    double* p_Xw  = new double [ns];
-    
+    Eigen::VectorXd p_Yke (ne);
```

Comment:
Instead of using Eigen here, you can use an std::vector<double>.  This is more sensible because basically you are just replacing the pointers and don't actually need any Eigen-specific features.  I would suggest to change like so:
```suggestion
    std::vector<double> Yke(ne, 0.0);
```
And then you don't need the `Yke.setZero()` lines below.  Obviously this would apply to the other Eigen::VectorXds...